xdsl/dialects/builtin.py

```diff
@@ -575,28 +575,28 @@ def from_int_list(
 
     @staticmethod
     @builder
-    def from_float_list(
-            type: VectorOrTensorOf[AnyFloat],
-            data: List[float | AnyFloatAttr]) -> DenseIntOrFPElementsAttr:
+    def create_dense_float(
+        type: VectorOrTensorOf[AnyFloat],
+        data: List[int | AnyIntegerAttr | float | AnyFloatAttr]
+    ) -> DenseIntOrFPElementsAttr:
         data_attr = [
-            FloatAttr.from_value(d, type.element_type)
-            if isinstance(d, float) else d for d in data
+            FloatAttr.from_value(float(d), type.element_type)
+            if not isinstance(d, FloatAttr) else d for d in data
         ]
         return DenseIntOrFPElementsAttr([type, ArrayAttr.from_list(data_attr)])
 
     @staticmethod
     @builder
     def from_list(
-        type: VectorOrTensorOf[Attribute],
-        data: List[int | IntegerAttr[IntegerType]]
-        | List[int | IntegerAttr[IndexType]] | List[float | AnyFloatAttr]
+        type: VectorOrTensorOf[Attribute], data: List[int | AnyIntegerAttr]
+        | List[float | AnyFloatAttr]
```

Comment:
Actually, thinking about it again, shouldn't this also allow int and AnyIntegerAttr? This is the case that gets forwarded to `create_dense`, so we explicitely want to allow mixed lists, right?

Reply:
Yes, this should also add `int`.